astropy/coordinates/baseframe.py

```diff
@@ -167,6 +168,7 @@ def __new__(cls, reprname, framename, defaultunit="recommended"):
 """
 
 
+@dataclass
```

Comment:
Why is this here???

Reply:
Ok, good catch. That one is there entirely because I was testing something for https://github.com/astropy/astropy/pull/15259. I'll remove it and we will squash-merge this PR.

Reply:
😆